HDS-2246 Delete old references to Showcase

KristinLBradley opened the pull request:
### :pushpin: Summary
If merged, this PR removes commented out links and references to Showcase in the web docs

<!-- 
### :hammer_and_wrench: Detailed description
If more details are appropriate, add them here. What code changed, and why?

### :camera_flash: Screenshots
Screenshots always help, especially if this PR will change what renders to the browser -->

### :link: External links

<!-- Issues, RFC, etc. -->
Jira ticket: [HDS-2246](https://hashicorp.atlassian.net/browse/HDS-2246)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed
- [ ] Confirm that A11y tests have been run locally for this component

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

jorytindall reviewed (approved):
Nice! 🎉

didoo commented:
@KristinLBradley @Dhaulagiri there is also a lot of JS code that in theory should be removed (it was there with the idea that would have been used by the "showcase" tab).

Just some examples:
- https://github.com/hashicorp/design-system/blob/main/website/docs/components/dropdown/index.js#L13-L29
- https://github.com/hashicorp/design-system/blob/main/website/docs/components/button/index.js#L15C1-L26
- https://github.com/hashicorp/design-system/blob/main/packages/components/tests/dummy/app/routes/components/form/masked-input.js#L8-L16

Essentially, one has to go through each backing class for the documentation pages (surely components, maybe even foundations) and remove this extra JS (and in some cases remove entirely the file).

Don't know if you want to do it in the context of this ticket or have a follow-up one. I'll leave it to you to decide and prioritize accordingly.

KristinLBradley commented:
> @KristinLBradley @Dhaulagiri there is also a lot of JS code that in theory should be removed (it was there with the idea that would have been used by the "showcase" tab).
> 
> Just some examples:
> 
> * https://github.com/hashicorp/design-system/blob/main/website/docs/components/dropdown/index.js#L13-L29
> * https://github.com/hashicorp/design-system/blob/main/website/docs/components/button/index.js#L15C1-L26
> * https://github.com/hashicorp/design-system/blob/main/packages/components/tests/dummy/app/routes/components/form/masked-input.js#L8-L16
> 
> Essentially, one has to go through each backing class for the documentation pages (surely components, maybe even foundations) and remove this extra JS (and in some cases remove entirely the file).
> 
> Don't know if you want to do it in the context of this ticket or have a follow-up one. I'll leave it to you to decide and prioritize accordingly.

I went ahead and deleted JS that seemed to be needed only by the Showcase. I left items such as console.log actions that seemed like they may be used in the Code tab examples. (Please let me know if any of the items I left should be deleted too though.)

MelSumner commented on website/docs/components/tabs/index.js:
```diff
@@ -6,23 +6,7 @@
-    return ['default', 'hover', 'active', 'focus'];
-  }
-
-  get COLORS() {
-    return COLORS;
-  }
-
-  @action
-  toggleHighlight() {
-    this.showHighlight = !this.showHighlight;
-  }
```
We don't still need these?

KristinLBradley replied:
The toggleHighlight action is used in the separate Showcase and is still working there. This one isn't being used in the Web site.

I couldn't find where the colors were being used for anything.

MelSumner reviewed (approved):
left a question but otherwise looks good! Thanks for working on this! 👍

didoo commented:
> I went ahead and deleted JS that seemed to be needed only by the Showcase. I left items such as console.log actions that seemed like they may be used in the Code tab examples. (Please let me know if any of the items I left should be deleted too though.)

@KristinLBradley there is some extra JS still to remove. My suggestion is to open one by one the JS files in the  `website/docs/components/**/index.js` folder and subfolders, and check the code for things like `TYPES`, `SIZES`, `COLORS`, `BACKGROUNDS`, etc. (usually are all uppercase, but not only) that are declared in the JS side, but not only that (in some cases there may also be functions, and we need to check if they're used or not). That's why you have to do it manually, one by one.

KristinLBradley commented:
> > I went ahead and deleted JS that seemed to be needed only by the Showcase. I left items such as console.log actions that seemed like they may be used in the Code tab examples. (Please let me know if any of the items I left should be deleted too though.)
> 
> @KristinLBradley there is some extra JS still to remove. My suggestion is to open one by one the JS files in the `website/docs/components/**/index.js` folder and subfolders, and check the code for things like `TYPES`, `SIZES`, `COLORS`, `BACKGROUNDS`, etc. (usually are all uppercase, but not only) that are declared in the JS side, but not only that (in some cases there may also be functions, and we need to check if they're used or not). That's why you have to do it manually, one by one.

I just went through and opened each js file individually and deleted references to TYPES, etc and also deleted JS files entirely which contained only references like those.

didoo commented:
> I just went through and opened each js file individually and deleted references to TYPES, etc and also deleted JS files entirely which contained only references like those.

@KristinLBradley one last possible leftover? can you double check? https://github.com/hashicorp/design-system/blob/hds-2246-remove-showcase-references/website/docs/components/form/radio-card/index.js#L9

KristinLBradley commented:
> > I just went through and opened each js file individually and deleted references to TYPES, etc and also deleted JS files entirely which contained only references like those.
> 
> @KristinLBradley one last possible leftover? can you double check? https://github.com/hashicorp/design-system/blob/hds-2246-remove-showcase-references/website/docs/components/form/radio-card/index.js#L9

I checked and it was unneeded so I deleted, thanks.

didoo reviewed (approved):
✅
